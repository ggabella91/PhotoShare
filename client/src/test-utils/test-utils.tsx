import { FC, ReactElement } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { store } from '../redux/store';

interface WrapperProps {
  wrapperProps?: { route: string; location: string };
}

type CustomRenderOptions = RenderOptions & WrapperProps;

const WithReduxProvider: FC<WrapperProps> = ({ children, wrapperProps }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes>
          <Route path={wrapperProps?.route || '/'} element={children} />
        </Routes>
      </BrowserRouter>
    </Provider>
  );
};

const renderWithReduxAndRouter = (
  ui: ReactElement,
  options?: Omit<CustomRenderOptions, 'wrapper'>
) => {
  window.history.pushState({}, '', options?.wrapperProps?.location || '/');
  const customContainer = document.createElement('div');
  customContainer.id = 'root';

  return render(ui, {
    wrapper: (props) => (
      <WithReduxProvider {...props} wrapperProps={options?.wrapperProps} />
    ),
    container: document.body.appendChild(customContainer),
    ...options,
  });
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { renderWithReduxAndRouter as render };
