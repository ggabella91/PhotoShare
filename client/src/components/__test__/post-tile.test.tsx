import { render } from '../../test-utils/test-utils';
import PostTile from '../post-tile/post-tile.component';

it('expect to render a PostTile component', () => {
  const { container: postTile } = render(
    <PostTile
      fileString='hello'
      custRef={null}
      dataS3Key=''
      onClick={() => console.log('I was clicked!')}
    />
  );

  expect(postTile).toBeInTheDocument();
});
