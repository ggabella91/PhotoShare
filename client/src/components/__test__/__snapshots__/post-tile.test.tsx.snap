// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render a PostTile component 1`] = `
<div
  className="post-tile"
>
  <img
    alt="post-pic"
    className="post-tile-image"
    onClick={[Function]}
    src="data:image/jpeg;base64,hello"
  />
</div>
`;
